### PR TITLE
Tests: add tests for delegation

### LINKED ISSUE #17: Tests: add test for signing delegation
There are no tests for signing delegation.

Add tests that:
 - check signature
 - check display
 - check constraints

### PR DESCRIPTION
**Pull request by spalmer25**
- [x] test the delegation signing display
- [x] test the delegation signing constraint

fixes #17

### CODE REVIEW

**Review by spalmer25 — COMMENTED** [review]
Do not merge while MR target is not `main`

File: test/python/test_instructions.py [hunk 1]
```diff
@@ -585,6 +586,106 @@ def test_sign_level_authorized(
...
+    )
+
+    delegation = Delegation(
+        delegate=account.public_key_hash,
+        source=account.public_key_hash,
+    )
+
+    raw_delegation = delegation.forge()
+
+    if with_hash:
```
Comment (ajinkyaraj-23) [inline test/python/test_instructions.py:613]: Shouldn't it be oppsite? if with_hash is true, we should sign with hash.

Comment (spalmer25) [reply test/python/test_instructions.py]: Fixed here: https://github.com/trilitech/ledger-app-tezos-baking/pull/24/commits/25cdc8d279a03be5050f5da1e213410ab36001d3

**Review by ajinkyaraj-23 — APPROVED** [review]
LGTM

**Pull request merged**